cmd/process-user-data/update.go

```diff
@@ -106,6 +107,39 @@ func updateAgentConfig(cmd *cobra.Command, args []string) error {
 		agentConfig.AaKbcParams = config.AAKBCParams
 	}
 
+	if config.AuthJson != "" {
+		// Check if AaKbcParams is offline_fs_kbc
+		if strings.Contains(agentConfig.AaKbcParams, "offline_fs_kbc") {
+			// Create offline registry auth file
+			// Create the file
+			file, err := os.Create(offlineKbcAuthFile)
+			if err != nil {
+				return fmt.Errorf("failed to create file: %s", err)
+			}
+			defer file.Close()
+
+			_, err = file.WriteString(cloudinit.AuthJSONToResourcesJSON(string(config.AuthJson)))
+			if err != nil {
+				return fmt.Errorf("failed to write agent config file: %s", err)
+			}
+
+		} else if strings.Contains(agentConfig.AaKbcParams, "cc_kbc") {
```

Comment:
I'm wondering if we should just use the `image_registry_auth_file=file:///etc/auth.json` config option for all kbcs so we have less branching and more shared code as it should still be valid?

Reply:
Does it apply for all KBCs ? I'm open for changing

Reply:
So my understanding (which could be wrong), is that by specifying `image_registry_auth_file=file:///etc/auth.json` image-rs uses that file directly, rather than going to the attestation-agent (and therefore KBC) to get that information. I'm happy to test it out trying to build a peer-pod with offline_fs_kbc if you make the code general